test/extensions/stats_sinks/dog_statsd/config_test.cc

```diff
@@ -65,6 +65,59 @@ TEST(DogStatsdConfigTest, ValidateFail) {
       ProtoValidationException);
 }
 
+TEST_P(DogStatsdConfigLoopbackTest, CustomBufferSize) {
+  const std::string name = StatsSinkNames::get().DogStatsd;
+
+  envoy::config::metrics::v3::DogStatsdSink sink_config;
+  sink_config.set_max_bytes_per_datagram(128);
+  envoy::config::core::v3::Address& address = *sink_config.mutable_address();
+  envoy::config::core::v3::SocketAddress& socket_address = *address.mutable_socket_address();
+  socket_address.set_protocol(envoy::config::core::v3::SocketAddress::UDP);
+  auto loopback_flavor = Network::Test::getCanonicalLoopbackAddress(GetParam());
+  socket_address.set_address(loopback_flavor->ip()->addressAsString());
+  socket_address.set_port_value(8125);
+
+  Server::Configuration::StatsSinkFactory* factory =
+      Registry::FactoryRegistry<Server::Configuration::StatsSinkFactory>::getFactory(name);
+  ASSERT_NE(factory, nullptr);
+
+  ProtobufTypes::MessagePtr message = factory->createEmptyConfigProto();
+  TestUtility::jsonConvert(sink_config, *message);
+
+  NiceMock<Server::MockInstance> server;
+  Stats::SinkPtr sink = factory->createStatsSink(*message, server);
+  ASSERT_NE(sink, nullptr);
+  auto udp_sink = dynamic_cast<Common::Statsd::UdpStatsdSink*>(sink.get());
```

Comment:
FYI this use of `auto` is correct as it's clear from C++ rules what the type of it must be, just from looking at the one line.